core/server/lib/members/static/auth/index.js

```diff
@@ -376,20 +411,32 @@ export default class App extends Component {
                     <a className="gm-modal-close" onClick={ (e) => this.close(e)}>{ IconClose }</a>
                     {this.renderFormHeaders(formType)}
                     {this.renderFormSection(formType)}
+                    {this.renderFormFooters(formType)}
                 </div>
             </div>
         );
     }
 
     render() {
         return (
-            <div className="gm-page-overlay" onClick={(e) => this.close(e)}>
+            <div className={this.state.parentContainerClass} onClick={(e) => this.close(e)}>
                 {this.renderFormComponent()}
             </div>
         );
     }
 
     close(event) {
-        window.parent.postMessage('pls-close-auth-popup', '*');
+        this.setState({
+            parentContainerClass: 'gm-page-overlay close'
+        });
+
+        window.setTimeout(function(){
```

Comment:
Elsewhere in this file you've used "arrow functions" (`() => {}`), which IMO are a lot neater - I wouldn't usually comment on a code style thing, but in this case, an arrow function would be really cool - because it preserves the value of `this` from the outer function

That would mean you don't need to use `.bind(this)` a few lines below.

It's not a big deal tho - feel free to leave it :+1:

Reply:
@allouis Thanks for the correction. Could you please doublecheck if it's good now? Thanks!

Reply:
:ok_hand: looks great